internal/models/factor.go

```diff
@@ -112,24 +119,45 @@ func ParseAuthenticationMethod(authMethod string) (AuthenticationMethod, error)
 		return TokenRefresh, nil
 	case "mfa/sms":
 		return MFAPhone, nil
+	case "mfa/webauthn":
+		return MFAWebAuthn, nil
 	}
 	return 0, fmt.Errorf("unsupported authentication method %q", authMethod)
 }
 
 type Factor struct {
 	ID uuid.UUID `json:"id" db:"id"`
 	// TODO: Consider removing this nested user field. We don't use it.
-	User             User               `json:"-" belongs_to:"user"`
-	UserID           uuid.UUID          `json:"-" db:"user_id"`
-	CreatedAt        time.Time          `json:"created_at" db:"created_at"`
-	UpdatedAt        time.Time          `json:"updated_at" db:"updated_at"`
-	Status           string             `json:"status" db:"status"`
-	FriendlyName     string             `json:"friendly_name,omitempty" db:"friendly_name"`
-	Secret           string             `json:"-" db:"secret"`
-	FactorType       string             `json:"factor_type" db:"factor_type"`
-	Challenge        []Challenge        `json:"-" has_many:"challenges"`
-	Phone            storage.NullString `json:"phone" db:"phone"`
-	LastChallengedAt *time.Time         `json:"last_challenged_at" db:"last_challenged_at"`
+	User               User                `json:"-" belongs_to:"user"`
+	UserID             uuid.UUID           `json:"-" db:"user_id"`
+	CreatedAt          time.Time           `json:"created_at" db:"created_at"`
+	UpdatedAt          time.Time           `json:"updated_at" db:"updated_at"`
+	Status             string              `json:"status" db:"status"`
+	FriendlyName       string              `json:"friendly_name,omitempty" db:"friendly_name"`
+	Secret             string              `json:"-" db:"secret"`
+	FactorType         string              `json:"factor_type" db:"factor_type"`
+	Challenge          []Challenge         `json:"-" has_many:"challenges"`
+	Phone              storage.NullString  `json:"phone" db:"phone"`
+	LastChallengedAt   *time.Time          `json:"last_challenged_at" db:"last_challenged_at"`
+	WebAuthnCredential *WebAuthnCredential `json:"-" db:"web_authn_credential"`
+	WebAuthnAAGUID     *uuid.UUID          `json:"web_authn_aaguid,omitempty" db:"aaguid"`
+}
+
+type WebAuthnCredential struct {
+	webauthn.Credential
+}
+
+func (wc *WebAuthnCredential) Value() (driver.Value, error) {
+	return json.Marshal(wc)
+}
+
+func (wc *WebAuthnCredential) Scan(value interface{}) error {
+	bytes, ok := value.([]byte)
+	if !ok {
+		return fmt.Errorf("expected byte array but got %T", value)
+	}
+
+	return json.Unmarshal(bytes, wc)
```

Comment:
Likewise for checking if this might be string.

Also could you treat empty arrays and strings as null instead?